src/box/lua/config/configdata.lua

```diff
@@ -62,9 +63,120 @@ function methods.names(self)
         group_name = self._group_name,
         replicaset_name = self._replicaset_name,
         instance_name = self._instance_name,
+        replicaset_uuid = self._replicaset_uuid,
+        instance_uuid = self._instance_uuid,
     }
 end
 
+local function uuid_from_name(str)
+    local sha = digest.sha1_hex(str)
+    return sha:sub(1,8)..'-'..sha:sub(9,12)..'-'..sha:sub(13,16)..'-'..
+           '00'..sha:sub(17,18)..'-'..sha:sub(19,30)
+end
+
+local function instance_sharding(iconfig, instance_name)
+    local roles = instance_config:get(iconfig, 'sharding.roles')
+    if roles == nil then
+        return nil
+    end
+    assert(type(roles) == 'table' and #roles > 0)
```

Comment:
Nit: I wouldn't forbid the empty list of roles. I propose to interpret it similarly to `roles == nil`.

Reply:
Fixed.